src/mapbox/mapbox.js

```diff
@@ -224,6 +226,18 @@ export default class Mapbox {
   }
 
   // PRIVATE API
+  // delayed fireLoadEvent in case map is unmounted right after
+  _fireLoadEvent = () => {
+    this._timer = window.setTimeout(() => {
```

Comment:
I do not think we can avoid that error completely, as this `mapbox` will call the `props.onLoad`, and have no control what the user will do in `props.onLoad`.

Also when `props.onLoad` is already triggered in `_fireLoadEvent`, I do not know what is the appropriate way to cancel this `props.onLoad` call. So I add a timer here to delay the call in case the map is unmounted before loaded.